opentelemetry-api/src/opentelemetry/trace/span.py

```diff
@@ -38,6 +38,8 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         """Sets an Attribute.
 
         Sets a single Attribute with the key and value passed as arguments.
+
+        Attribute values of `None` are not valid and attempting to set a `None` value is undefined behavior.
```

Comment:
I recommend using the same language as the java implementation for consistency: https://github.com/open-telemetry/opentelemetry-java/blob/1512a537041b235c76ba3dceae02fa65d6751c68/api/src/main/java/io/opentelemetry/api/common/Attributes.java#L31

```suggestion
        Note: The behavior of `None` value attributes is undefined, and hence strongly discouraged.
```

Reply:
I picked the language from [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/common/common.md), but anyway it's good be consistent with other languages.